### PR TITLE
Closes #290: editFilesFragment_refactor

### LINKED ISSUE #290: Find and refactor usage of `dataversePage` param and `DataversePage` backing bean inside `editFilesFragment.xhtml`

### PR DESCRIPTION
**Pull request by dbojanek**
Some parameters in editdatafiles.xhtml and dataset.xhtml are redundant, but I think leaving it at this state will make future refactor easier.

### CODE REVIEW

File: dataverse-webapp/src/main/webapp/editFilesFragment.xhtml [hunk 1]
```diff
@@ -396,7 +373,7 @@
                                 <p:message for="fileDescription"/>
                             </ui:fragment>
                         </div>
-                        <div jsf:rendered="#{createDataset or EditDatafilesPage.mode eq 'UPLOAD' }">
+                        <div jsf:rendered="#{fileTermsOfUse}">
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editFilesFragment.xhtml]: I think name/param should be more descriptive, like `isUploadMode`;

File: dataverse-webapp/src/main/webapp/editFilesFragment.xhtml [hunk 2]
```diff
@@ -131,8 +128,8 @@
                     <p:fileUpload id="fileUpload" 
                                   dragDropSupport="true"
                                   auto="true"
-                                  multiple="#{datasetPage || EditDatafilesPage.allowMultipleFileUpload()}"
-                                  disabled="#{lockedFromEdits ||  !(datasetPage || EditDatafilesPage.showFileUploadComponent()) }"
+                                  multiple="#{fileUploadMultiple}"
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editFilesFragment.xhtml]: Maybe `multipleUploadEnabled`?

File: dataverse-webapp/src/main/webapp/editFilesFragment.xhtml [hunk 3]
```diff
@@ -43,7 +40,7 @@
                 </ui:fragment>
                 <!-- Upload -->
                 <div class="panel-group" role="tablist" aria-multiselectable="true"
-                      id="fileUploadFragment" jsf:rendered="#{datasetPage || EditDatafilesPage.showFileUploadFragment()}">
+                      id="fileUploadFragment" jsf:rendered="#{fileUploadFragment}">
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editFilesFragment.xhtml]: Maybe `showFileUpload`?

File: dataverse-webapp/src/main/webapp/editFilesFragment.xhtml [hunk 4]
```diff
@@ -322,12 +299,12 @@
                              selection="#{EditDatafilesPage.selectedFiles}" 
                              var="fileMetadata" 
                              widgetVar="filesTable"
-                             emptyMessage="#{datasetPage || EditDatafilesPage.showFileUploadFragment() ? bundle['file.noUploadedFiles.tip'] : bundle['file.noSelectedFiles.tip']}">
+                             emptyMessage="#{filesTableEmptyMessage ? bundle['file.noUploadedFiles.tip'] : bundle['file.noSelectedFiles.tip']}">
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editFilesFragment.xhtml]: Maybe `showFileUpload`?

File: dataverse-webapp/src/main/webapp/editdatafiles.xhtml [hunk 5]
```diff
@@ -60,10 +57,17 @@
 
                     <ui:include src="editFilesFragment.xhtml">
                         <ui:param name="editDatafilesPage" value="true"/>
+                        <ui:param name="datasetPage" value="false"/>
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: I'm not sure what does this parameter mean, when should it be false/true? What does it control?

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/webapp/editFilesFragment.xhtml (hunk 1 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/editFilesFragment.xhtml]: I've coupled params names with ids of elements they are used in. They idea was for them to be easily tracked during separation of Files tab.
But you're right it'll be better to give them more meaningful names.
We should however give names that are meaningful from both `EditDatafilesPage` and `DatasetPage` thus in this case `DatasetPage` is not aware of `mode eq 'UPLOAD'` so the name should be more like: `isUploadable` or `isUploadPermited` or such.

File: dataverse-webapp/src/main/webapp/editdatafiles.xhtml (hunk 5 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: I've added explicit parameter to have it coupled with the one on dataset.xhtml. During separation of `Files` tab we'll have explicit information that section rendered with `datasetPage` could be removed from editdatafiles.xhtml perspective.
It serves the same purpose as `editDatafilesPage` parameter.

Comment (MrK191) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: Maybe it should be named something like `renderDatasetPageSection`?

Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: Everywhere in the app we use `datasetPage` to reference anything that happens on/comes from dataset.xhmt.
I would leave it as it is.

Comment (MrK191) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: If something is badly named when is it the time to change it if not now?

Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml]: This param will be removed on separation from dataset.xhtml.
But if you insist then no problem.

File: dataverse-webapp/src/main/webapp/editdatafiles.xhtml [hunk 6]
```diff
@@ -60,10 +57,16 @@
...
+                        <ui:param name="isDatasetSection" value="false"/>
                         <ui:param name="createDataset" value="false"/>
                         <ui:param name="showFileButtonUpdate" value="true"/>
                         <ui:param name="version" value="#{EditDatafilesPage.workingVersion}"/>
                         <ui:param name="lockedFromEdits" value="#{EditDatafilesPage.lockedFromEdits}"/>
+                        <ui:param name="showFileUpload" value="#{EditDatafilesPage.showFileUploadFragment()}"/>
+                        <ui:param name="isMultiplyFilesUploadAllowed" value="#{EditDatafilesPage.allowMultipleFileUpload()}" />
+                        <ui:param name="fileUploadDisabled" value="#{!EditDatafilesPage.showFileUploadComponent()}" />
+                        <ui:param name="isUploadPermited" value="#{EditDatafilesPage.mode eq 'UPLOAD'}" />
+                        <ui:param name="redirectToEditdataFiles" value="" />
```
Comment (MrK191) [inline dataverse-webapp/src/main/webapp/editdatafiles.xhtml:69]: If redirect is not used in this view then I 'think' you can remove param.

Comment (dbojanek) [reply dataverse-webapp/src/main/webapp/editdatafiles.xhtml:69]: It is used with specific value of ""

**Review by MrK191 — APPROVED** [review]

**Pull request merged**